consul/base.py

```diff
@@ -1928,146 +1928,117 @@ def list(self, token=None):
-            if token_type:
-                assert token_type in ("client", "management"), "token_type must be client or management"
-                payload["Type"] = token_type
-            if rules:
-                assert isinstance(rules, str), "Only HCL or JSON encoded strings supported for the moment"
-                payload["Rules"] = rules
-
-            data = json.dumps(payload)
-
-            return self.agent.http.put(CB.json(is_id=True), "/v1/acl/update", params=params, data=data)
+            json_data = {"Description": description}
+            return self.agent.http.put(
+                CB.json(),
+                f"/v1/acl/token/{accessor_id}/clone",
+                params=params,
+                data=json.dumps(json_data),
+            )
 
-        def clone(self, acl_id, token=None):
+        def create(self, token=None, accessor_id=None, secret_id=None, description=""):
             """
-            Clones the ACL token *acl_id*. This is a privileged endpoint, and
-            requires a management token. *token* will override this client's
-            default token. An *ACLPermissionDenied* exception will be raised if
-            a management token is not used.
-
-            Returns the string of the newly created *acl_id*.
+            Create a token (optionally identified by *secret_id* and *accessor_id*).
+            This is a privileged endpoint, and requires a token with acl:write.
+            :param token: token with acl:write capability
+            :param accessor_id: The accessor ID of the token to create
+            :param secret_id: The secret ID of the token to create
+            :param description: Optional new token description
+            :return: The cloned token information
             """
             params = []
             token = token or self.agent.token
             if token:
                 params.append(("token", token))
-            return self.agent.http.put(CB.json(is_id=True), f"/v1/acl/clone/{acl_id}", params=params)
 
-        def destroy(self, acl_id, token=None):
-            """
-            Destroys the ACL token *acl_id*. This is a privileged endpoint, and
-            requires a management token. *token* will override this client's
-            default token. An *ACLPermissionDenied* exception will be raised if
-            a management token is not used.
+            json_data = {}
+            if accessor_id:
+                json_data["AccessorID"] = accessor_id
+            if secret_id:
+                json_data["SecretID"] = secret_id
+            if description:
+                json_data["Description"] = description
+            return self.agent.http.put(
+                CB.json(),
+                "/v1/acl/token",
+                params=params,
+                data=json.dumps(json_data),
+            )
 
-            Returns *True* on success.
+        def update(self, accessor_id, token=None, secret_id=None, description=""):
```

Comment:
I find it strange to want to update the secret_id

Reply:
Taken out from the endpoint doc :

[AccessorID](https://developer.hashicorp.com/consul/api-docs/acl/tokens#accessorid-3) (string: "") - Specifies the accessor ID of the token being updated. This field is required in the URL path but may also be specified in the JSON body. If specified in both places then they must be an exact match. This field is immutable. If not present in the body and only in the URL then it will be filled in by Consul.

[SecretID](https://developer.hashicorp.com/consul/api-docs/acl/tokens#secretid-1) (string: "") - Specifies the secret ID of the token being updated. This field is immutable so if present in the body then it must match the existing value. If not present then the value will be filled in by Consul.